### PR TITLE
[FLINK-33543][core] Move Java record related tests into their own module since they only compile under JDK17

### PR DESCRIPTION
**Pull request by XComp**
## What is the purpose of the change

It appears that the way Java record support is introduced in FLINK-32380 doesn't work well together with Intellij. The class exclusion isn't picked up by Intellij. That causes issues when trying build `flink-core` with Java 8 or 11. Moving the test classes into their own module solves the issue.

Ideally, `flink-java17` should rather depend on a more fine-grained module (e.g. flink-serialization). But splitting `flink-core` into smaller modules seems a bit out-of-scope and should be handled in a dedicated issue [as suggested by Chesnay](https://issues.apache.org/jira/browse/FLINK-30774?focusedCommentId=17680167&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17680167).

## Brief change log

* Moves Java record related test code into `flink-java17` module
* Removes `Java17` prefix from test class names

## Verifying this change

* CI test run with:
  * [Java 8](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=54558&view=logs&j=fc5181b0-e452-5c8f-68de-1097947f6483&t=995c650b-6573-581c-9ce6-7ad4cc038461) (no `flink-tests-java17` module included)
  * [Java 11](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=54558&view=logs&j=c300857e-4641-5071-7bcc-4860f446f6ac&t=ee1cb502-e57a-5f16-6ec1-5c1304b058fe) (no `flink-tests-java17` module included)
  * [Java 17](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=54558&view=logs&j=4572530b-40bd-5aff-0832-039ac543dd70&t=3db1abd1-9a91-57c4-025d-942bad9e4c4e&l=32911) Java 17 module build and test are executed

All test runs were based on [this build](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=54558). The test failure is unrelated to the change.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: yes
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1088,6 +1087,10 @@ under the License.
 				<scala.version>2.12.15</scala.version>
 			</properties>
 
+			<modules>
+				<module>flink-tests-java17</module>
+			</modules>
```
Comment (gyfora) [inline pom.xml:1092]: Shouldn't this belong under java17-target instead? The new module won't compile with other target java versions

File: tools/maven/suppressions-tests-java17.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+<!DOCTYPE suppressions PUBLIC
+	"-//Puppy Crawl//DTD Suppressions 1.1//EN"
+	"http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+	<!-- Java records aren't supported by checkstyle 8.14. The corresponding check error needs to be suppressed. -->
```
Comment (zentol) [inline tools/maven/suppressions-tests-java17.xml:26]: Good job documenting the actual reason for the exclusion 👍

**Review by zentol — COMMENTED** [review]
Looks good minus Gyulas comment regarding java17-target.

File: pom.xml (hunk 1 above)
Comment (XComp) [reply pom.xml]: :+1: true - I always mix them up :facepalm:

**Review by gyfora — APPROVED** [review]

**Comment by XComp** [comment]
Thanks for the approval. I squashed the commits and removed the debug commit (which modified the CI run).

**Pull request merged**